plugins/serializers/binary/entry_test.go

```diff
@@ -0,0 +1,285 @@
+package binary
+
+import (
+	"encoding/binary"
+	"github.com/stretchr/testify/require"
+	"math"
+	"testing"
+	"time"
+)
+
+func TestIntSerialization(t *testing.T) {
+	testData := 1
+	expected := []byte{0x00, 0x00, 0x00, 0x01}
+
+	e := &Entry{Name: "test", DataFormat: "int32"}
+	errCheck := e.Check()
+	value, errSerialize := e.SerializeValue(testData, binary.BigEndian)
+
+	require.NoError(t, errCheck)
+	require.NoError(t, errSerialize)
+
+	require.Equal(t, expected, value)
+}
+
+func TestNegativeIntSerialization(t *testing.T) {
```

Comment:
All these separate tests doing similar checks can maybe be combined into 1 and run these as multiple test cases?